### PR TITLE
[EAM-1702] Equipment tree state rework

### PR DESCRIPTION
**Pull request by MarbleToast**

### CODE REVIEW

File: src/ui/components/eqtree/EAMTree.js [hunk 1]
```diff
@@ -1,205 +1,190 @@
...
+            if (error.type !== ErrorTypes.REQUEST_CANCELLED) {
+                handleError(error);
+            }
+        }
+    };
+
+    useEffect(() => {
+        //console.log('HEY');
+        //console.log(eqpTreeNewChild);
+        //console.log(eqpTreeNewRoot);
```
Comment (skamv) [inline src/ui/components/eqtree/EAMTree.js:56]: We should remove commented console.logs to keep the codebase clean

**Comment by skamv** [comment]
Also I would rebase and fix/squash some commits and add more descriptive messages to some of them before merging

File: src/ui/layout/menu/EamlightMenu.js [hunk 2]
```diff
@@ -189,6 +191,16 @@ class EamlightMenu extends Component {
...
+
+                        {reports && reports.length > 0 &&
+                        <li>
+                            <div rel="customgrids" onClick={this.mainMenuClickHandler}>
+                                <Tooltip title="CUSTOM GRIDS" placement="right">
+                                    <FormatListBulletedTriangle style={iconStyles} />
+                                </Tooltip>
+                            </div>
+                        </li>
+                        }
```
Comment (ricardo-j-almeida) [inline src/ui/layout/menu/EamlightMenu.js:203]: This should not be part of this PR as it is a completely different feature

File: src/ui/layout/menu/MenuGridLink.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    return (
+        <li>
+            <Link to={"/grid?gridName=" + props.grid.code} style={linkStyle}>
+                <span style={{...spanStyle, display: 'inline', paddingLeft: 5}}>{props.grid.desc}</span>
+            </Link>
+        </li>
+    )
+}
+
+export default MenuGridLink;
```
Comment (ricardo-j-almeida) [inline src/ui/layout/menu/MenuGridLink.js:26]: This should not be part of this PR as it is a completely different feature

File: src/ui/pages/Entity.js [hunk 4]
```diff
@@ -318,6 +330,8 @@ export default class readEntityEquipment extends Component {
             .then(response => {
                 this.props.showNotification(this.settings.entityDesc + ' ' + code + ' has been successfully deleted.')
                 this.props.history.push(this.settings.entityURL)
+                // Change the window title
```
Comment (ricardo-j-almeida) [inline src/ui/pages/Entity.js:333]: This should not be part of this PR as it is a completely different feature

**Comment by MarbleToast** [comment]
A lot of changes needed for this PR were tangled in old commits for another unrelated commit. I have now gone through, separated, and squashed relevant changes.

File: src/ui/pages/equipment/installeqp/InstallEqp.js [hunk 5]
```diff
@@ -1,74 +1,134 @@
...
+        if (parentEq) {
+            params += `parent=${parentEq}`;
+        }
+        if (childEq) {
+            if (params !== '?') params += `&`;
+            params += `child=${childEq}`;
+        }
+        if (params === '?') {
+            params = '';
+        }
```
Comment (skamv) [inline src/ui/pages/equipment/installeqp/InstallEqp.js:59]: why are we not relying on queryString to generate the params string? What if we have some other params? They should not be joined with the parent and child?
